### DISCUSSION TITLE
SSR  elment Plus<title>

### DISCUSSION THREAD

**Discussion by SmileLiyan** [community/community#127320] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Why does the table component of element Plus not support server-side rendering?I am currently using element plus in my project, but only the table component of element plus does not support SSR，how can I solve this problem?

**Comment by InboraStudio** [comment]
The Element Plus table component doesn't support server-side rendering (SSR) because it relies on browser-specific features and APIs that aren't available during SSR.

Wrap the table component in a ClientOnly component to ensure it only renders on the client side.

```vue
<template>
  <ClientOnly>
    <el-table :data="tableData">
      <!-- table columns here -->
    </el-table>
  </ClientOnly>
</template>

<script>
export default {
  data() {
    return {
      tableData: [] // your data fetching logic here
    };
  },
  created() {
    if (process.client) {
      this.fetchTableData();
    }
  },
  methods: {
    async fetchTableData() {
      // fetch data logic here
      this.tableData = await fetchDataFromAPI();
    }
  }
};
</script>

```

**Reply by SmileLiyan** [reply]
thank you for your reply, our company must do ssr ,so I used the antd table
